### PR TITLE
[ARM][Thumb2] Allow 2-operand variants of `[us]div`

### LINKED ISSUE #119963: Clang doesn't accept udiv with 2 operands when assembling for Cortex-M3
When using clang to assemble for ARM Cortex-M3, it doesn't accept an `udiv` instruction with 2 operands, for example:

```asm
.syntax unified

udiv   r10, r5
```

An attempt to assemble that using clang v19.1.5:

```
clang -nodefaultlibs -mcpu=cortex-m3 --target=armv7m-none-eabi ./example.s
```

results in an error:

```
./example.s:3:15: error: too few operands for instruction
udiv   r10, r5
              ^
```

However, *Arm v7-M Architecture Reference Manual* says that the assembler syntax is `UDIV<c><q> {<Rd>,} <Rn>, <Rm>` where *"`<Rd>` Specifies the destination register. If `<Rd>` is omitted, this register is the same as `<Rn>`."* As a result, I expect clang to assemble that into:
```
fbba faf5   udiv r10, r10, r5
```
GCC assembles it correctly: https://godbolt.org/z/oh8TnW99T.

### PR DESCRIPTION
**Pull request by chestnykh**
Fixes #119963

### CODE REVIEW

**Comment by s-barannikov** [comment]
Looks reasonable, but I'm not affiliated to ARM to approve.

File: llvm/test/MC/ARM/idiv-2op.s [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@ RUN: llvm-mc -triple=thumbv7 -mcpu=cortex-a15 -show-encoding < %s 2>&1 | FileCheck -check-prefix A15-THUMB %s
+
+@ RUN: llvm-mc -triple=armv7 -mcpu=cortex-a15 -mattr=-hwdiv -show-encoding < %s 2>&1 | FileCheck -check-prefix A15-ARM-NOTHUMBHWDIV %s
+@ RUN: llvm-mc -triple=thumbv7 -mcpu=cortex-a15 -mattr=-hwdiv-arm -show-encoding < %s 2>&1 | FileCheck -check-prefix A15-THUMB-NOARMHWDIV %s
+
+@ RUN: llvm-mc -triple=armv8 -show-encoding < %s 2>&1 | FileCheck -check-prefix ARMV8 %s
+@ RUN: llvm-mc -triple=thumbv8 -show-encoding < %s 2>&1 | FileCheck -check-prefix THUMBV8 %s
+
+@ RUN: llvm-mc -triple=armv8 -mattr=-hwdiv -show-encoding < %s 2>&1 | FileCheck -check-prefix ARMV8-NOTHUMBHWDIV %s
+@ RUN: llvm-mc -triple=thumbv8 -mattr=-hwdiv-arm -show-encoding < %s 2>&1 | FileCheck -check-prefix THUMBV8-NOTHUMBHWDIV %s
```
Comment (ostannard) [inline llvm/test/MC/ARM/idiv-2op.s:14]: Minor nit: this check prefix should be `THUMBV8-NOARMHWDIV`

**Review by ostannard — APPROVED** [review]
LTGM too, with one minor nit.

File: llvm/test/MC/ARM/idiv-2op.s (hunk 1 above)
Comment (chestnykh) [reply llvm/test/MC/ARM/idiv-2op.s]: Thanks, fixed in this file and in the original test source

**Pull request merged**